feat: reduce validator operation logging noise with aggregated summaries

diegomrsantos opened the pull request:
## Issue Addressed

This PR addresses excessive logging noise in anchor.log during validator synchronization operations. No specific GitHub issue was filed, but the problem was identified through log analysis showing thousands of repetitive DEBUG messages during bulk validator processing.

## Proposed Changes

- **anchor/eth/src/event_processor.rs**: 
  - Move individual "Successfully added validator" and "Successfully removed validator" logs from DEBUG to TRACE level
  - Add simple counters to track validator operations per batch
  - Add summary DEBUG logs showing total validators added/removed after processing each batch
- **CLAUDE.md**: 
  - Add "Simplicity First" principle to general development guidelines
  - Add "Check Requirements First" principle to prevent template/guideline oversights
  - Add guidance for updating documentation after successful sessions
  - Emphasize reading PR template before creating PRs
- **.claude/agents/logging-subagent.md**: 
  - Add logging-specific principles for noise reduction strategies
  - Document preference for simple aggregation over complex data structures

## Additional Info

This change dramatically reduces log noise during validator synchronization while preserving all detailed information at TRACE level. The implementation uses simple counters rather than complex collections, following the principle of choosing the simplest solution that elegantly solves the problem. Operators will see clean summaries like "Added validators count=150" instead of 150 individual DEBUG messages, while detailed logs remain available when RUST_LOG=trace is enabled.

diegomrsantos committed:
feat(logging): reduce validator operation log noise

- Change individual validator add/remove logs from DEBUG to TRACE level
- Add aggregated summary logging showing total validators added/removed per batch
- Maintains detailed logs at TRACE level while providing clean operational visibility

diegomrsantos committed:
chore: stage logging-subagent changes for rebase

diegomrsantos committed:
docs: update guidance with simplicity principles and logging best practices

- Add 'Simplicity First' principle to CLAUDE.md general guidelines
- Add session learning update guidance for future improvements
- Enhance logging-subagent with noise reduction strategies
- Document preference for simple counters over complex collections
- Establish patterns for TRACE vs DEBUG level logging decisions

diegomrsantos committed:
Merge branch 'release-v0.3.0' into feat/logging-noise-reduction

diegomrsantos committed:
update claude.md

dknopik commented on anchor/eth/src/event_processor.rs:
```diff
@@ -97,10 +101,12 @@ impl EventProcessor {
                 }
 
                 SSVContract::ValidatorRemoved::SIGNATURE_HASH => {
+                    validators_removed += 1;
```
These should only be incremented on success IMO

diegomrsantos replied:
done

diegomrsantos committed:
fix: only count successfully processed validator operations

- Use Result::inspect to increment counters only on successful operations
- Ensures logs accurately reflect validators actually added/removed
- Prevents misleading counts when event processing fails

diegomrsantos committed:
Merge branch 'release-v0.3.0' into feat/logging-noise-reduction

dknopik reviewed (approved):
LGTM
